Framework/Algorithms/test/ConvertSpectrumAxis2Test.h

```diff
@@ -468,6 +468,22 @@ class ConvertSpectrumAxis2Test : public CxxTest::TestSuite {
       TS_ASSERT_DELTA((*axis)(i), double(i % 10), 1E-10);
     }
   }
+
+  void test_eventWS() {
+    const std::string outputWS("outWS");
+    const std::string target("theta");
+    auto testWS =
+        WorkspaceCreationHelper::createEventWorkspaceWithFullInstrument(500, 3);
+    Mantid::Algorithms::ConvertSpectrumAxis2 conv;
+    conv.initialize();
```

Comment:
Because you haven't set `conv.setRethrows(true)`, the `execute()` below will not throw anyway, so the ASSERT check wrapping it is redundant.

IMHO, TS_ASSERT_THROWS_NOTHING is usually noise and acutally makes it harder to establish the point of failure in the test so I suggest removing all of them from here.

Reply:
You should call `conv.setChild(true)` that way the algorithm will not interact with the ADS, and you can get the output workspace easily without going through the ADS.